pkg/csi_driver/controller.go

```diff
@@ -104,28 +104,26 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 			return nil, status.Error(codes.AlreadyExists, err.Error())
 		}
 	} else {
-
-		// Get the reserved-IPV4-cidr from the request. If value is not provided, reservedIPV4Range is defaulted to ""
+		// If we are creating a new instance, we need to evaluate the reservedIPRange range if reserved-ipv4-cidr is provided
+		// If the param was not provided, we default reservedIPRange to "" and cloud provider takes care of the allocation
 		reservedIPV4CIDR, ok := req.GetParameters()[paramReservedIPV4CIDR]
 		if ok {
-
-			// Prevents lock acquisition in case the CIDR is invalid
 			validCIDR := s.config.ipAllocator.ValidateCIDR(reservedIPV4CIDR)
 			if !validCIDR {
-				return nil, err
+				return nil, fmt.Errorf("The provided reserved-ipv4-cidr %s evaluated to be invalid after parsing", reservedIPV4CIDR)
 			}
 
 			reservedIPRange, err := s.reserveIPV4Range(ctx, newFiler, reservedIPV4CIDR)
+
+			// Possible cases are 1) CreateInstanceAborted, 2)CreateInstance running in background
+			// In either cases, the ListInstances responses would consist of the updated result
+			defer s.config.ipAllocator.ReleaseIPV4Range(reservedIPRange)
```

Comment:
The naming of this function is symmetrical with the controller call, but not with the ipAllocator.  It may be cleaner to make it symmetrical with ipAlloactor, and break up reserveIPRange into "getUsedIPRanges" and then call the ipAllocator.GetUnreservedIPBlock.

Let's also try to standardize on the terminology.  We're using ipblock, iprange, and ipv4-cidr.  Maybe we can reduce it down to two.  

1. ipv4-cidr to refer to the bigger ipnet that we're extracting from.  This aligns with the rest of GKE naming
2. ipRange to refer to the /29 ipnet that we pass to Filestore.  This aligns with Filestore naming.